vm/systemSmartContracts/delegation.go

```diff
@@ -1498,6 +1498,7 @@ func (d *delegation) finishDelegateUser(
 		return vmcommon.UserError
 	}
 
+	d.eei.Finish(delegateValue.Bytes())
```

Comment:
these changes are not backwards compatible. It should not be targeted to RC-JULY as that RC is closed from features perspective. Backward compatibility flag has to be added. There in no need to add these information for normal users, only for SC calling, any information like this has to be limited.
For liquid staking - it is almost mandatory to claim rewards and delegate after, as redelegation can fail for various reasons. Separate delegation means more control of funds. For staking phase 4 we have a series of changes on staking, delegation, delegation manager, we might introduce a set of things there. feat/liquid-staking is the branch.